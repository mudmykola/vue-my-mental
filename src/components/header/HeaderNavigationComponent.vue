<template>
  <div class="header-menu">
    <ul class="flex items-center gap-5">
      <li v-for="item in menuItems" :key="item.title" @mouseenter="showDropdown(item)" @mouseleave="hideDropdown(item)">
        <a :href="item.link">{{ item.title }}</a>
        <span v-if="item.title === 'Спеціалістам'" class="menu-icon" :class="{ 'rotate-180': item.isDropdownVisible }">
          <IconAngelDown/>
</span>
        <ul v-if="item.isDropdownVisible" class="dropdown-menu ">
          <li class="w-full px-4" v-for="subItem in item.submenu" :key="subItem.title">
            <a :href="subItem.link">{{ subItem.title }}</a>
          </li>
        </ul>
      </li>
    </ul>
  </div>
</template>

<script setup>
import {ref} from 'vue';
import IconAngelDown from "@/components/icons/IconAngelDown.vue";

const menuItems = ref([
  {
    title: 'Про нас',
    link: '/about',
    isDropdownVisible: false,
  },
  {
    title: 'Наші фахівці',
    link: '/experts',
    isDropdownVisible: false,
  },
  {
    title: 'Спеціалістам',
    link: '/specialists',
    isDropdownVisible: false,
    submenu: [
      {
        title: 'Медицина',
        link: '/specialists/medicine',
      },
      {
        title: 'Техніка',
        link: '/specialists/tech',
      },
      {
        title: 'Право',
        link: '/specialists/law',
      }
    ],
  }
]);

function showDropdown(item) {
  item.isDropdownVisible = true;
}

function hideDropdown(item) {
  item.isDropdownVisible = false;
}
</script>

<style scoped>
.menu-icon {
  margin-left: 4px;
  transition: transform 0.3s ease;
  display: inline-block;
}

.rotate-180 {
  transform: rotate(180deg);
}

.dropdown-menu {
  position: absolute;
  background-color: #fff;
  box-shadow: 0 2px 12px rgba(0, 0, 0, 0.1);
  padding: 8px 0;
  z-index: 1000;
  display: none;
}

.header-menu li:hover .dropdown-menu {
  display: block;
}
</style>