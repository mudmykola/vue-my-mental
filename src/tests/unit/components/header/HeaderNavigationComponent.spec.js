import HeaderNavigationComponent from "@/components/header/HeaderNavigationComponent.vue";
import { mount } from "@vue/test-utils";

describe("HeaderNavigationComponent.vue", () => {
	let wrapper;

	beforeEach(() => {
		wrapper = mount(HeaderNavigationComponent);
	});

	it("renders correctly", () => {
		expect(wrapper.element).toMatchSnapshot();
	});

	it("should have dropdown menu element", async () => {
		const specialistMenuItem = wrapper.findAll(".header-menu__item").at(2);

		await specialistMenuItem.trigger("mouseenter");

		const dropdownMenu = specialistMenuItem.find(".dropdown-menu");
		expect(dropdownMenu.exists()).toBe(true);

		expect(dropdownMenu.isVisible()).toBe(true);
	});
});