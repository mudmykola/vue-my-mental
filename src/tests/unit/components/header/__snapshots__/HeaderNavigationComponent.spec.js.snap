// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderNavigationComponent.vue renders correctly 1`] = `
<div
  class="header-menu"
>
  <ul
    class="flex items-center gap-5"
  >
    
    <li
      class="header-menu__item"
    >
      <a
        class="text-[14px] text-primary font-bold font-sans"
        href="/about"
      >
        Про нас
      </a>
      <!--v-if-->
      <!--v-if-->
    </li>
    <li
      class="header-menu__item"
    >
      <a
        class="text-[14px] text-primary font-bold font-sans"
        href="/experts"
      >
        Наші фахівці
      </a>
      <!--v-if-->
      <!--v-if-->
    </li>
    <li
      class="header-menu__item"
    >
      <a
        class="text-[14px] text-primary font-bold font-sans"
        href="/specialists"
      >
        Спеціалістам
      </a>
      <span
        class="menu-icon ml-1 transition-transform duration-300 ease"
      >
        <svg
          fill="none"
          height="10"
          viewBox="0 0 10 10"
          width="10"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 3.35737L3.96667 7.31862C4.2405 7.59118 4.61114 7.74419 4.9975 7.74419C5.38386 7.74419 5.7545 7.59118 6.02833 7.31862L10 3.35279L9.11625 2.4707L5.145 6.43612C5.10593 6.47518 5.05295 6.49712 4.99771 6.49712C4.94247 6.49712 4.88948 6.47518 4.85042 6.43612L0.88375 2.47529L0 3.35737Z"
            fill="black"
            style="fill: black; fill-opacity: 1;"
          />
        </svg>
      </span>
      <!--v-if-->
    </li>
    
  </ul>
</div>
`;
